addons/hr/models/hr_employee_base.py

```diff
@@ -33,8 +33,7 @@ class HrEmployeeBase(models.AbstractModel):
     work_contact_id = fields.Many2one('res.partner', 'Work Contact', copy=False)
     related_contact_ids = fields.Many2many('res.partner', 'Related Contacts', compute='_compute_related_contacts')
     related_contacts_count = fields.Integer('Number of related contacts', compute='_compute_related_contacts_count')
-    work_location_id = fields.Many2one('hr.work.location', 'Work Location', compute="_compute_work_location_id", store=True, readonly=False,
-    domain="[('address_id', '=', address_id), '|', ('company_id', '=', False), ('company_id', '=', company_id)]")
+    work_location_id = fields.Many2one('hr.work.location', 'Work Location', store=True, readonly=False)
```

Comment:
store=True can be removed since the field is not computed anymore, it's True by default